misk/src/main/kotlin/misk/config/HealthCheck.kt

```diff
@@ -0,0 +1,39 @@
+package misk.config
+
+import com.google.common.util.concurrent.Service
+import com.google.common.util.concurrent.Service.State
+
+interface HealthCheck {
+    /**
+     * Returns true if the service is able to operate normally. For an HTTP service this indicates
+     * the ability to accept traffic, for a database service this would indicate a valid connection
+     * that can be used to query the database.
+     */
+    fun isReady(): Boolean
+
+    /**
+     * Returns true if the service has the ability to operate normally now or in the future. Returns
+     * false is the service has failed and requires a restart.
+     */
+    fun isAlive(): Boolean
```

Comment:
strongly prefer having this return a message with the check, similar to the way our internal status checks and dropwizard healthchecks work. that way tools and human observers can easily see exactly which checks are failing. also need to allow anyone to install health checks, not just base them on the state of a single running service.